### DISCUSSION TITLE
Maybe Feature?: Bookmark Folders & Permissions

### DISCUSSION THREAD

**Discussion by JugaadiTek** [directus/directus#21460] [Feature Requests]
## Summary

Bookmarks in the app should be organizable into folders, rather than all nested underneath the parent collection.

### How it is

![image](https://github.com/directus/directus/assets/106853815/5f839f30-094f-4f17-abcf-2fd64e2a968d)

### What I would like to be able to do

![image](https://github.com/directus/directus/assets/106853815/3c15612f-a8a5-454a-8fd2-5f051f2ff1e5)

### Bookmark Overwhelm.
I find "saved views" both a very important & highly used feature in many different use cases.
**Incomplete list of examples:** analytics, dataprocessing, Projection Models, auditing manually entered data, troubleshooting scripts, troubleshooting directus flows, etc...

**The Problem is:** The # of bookmarks beneath key collections can quickly spiral out of control/become overwhelming to sort through

### Keeping Users, Clients & Team Members on Rails
Keeping people on a specific track to accomplish a goal with minimal context switching & hiding away distractions can be very important.

## Detailed Design
_(I might come back and expand this)_

### Must Have:
- **Simplest Solution:** _(if movable bookmarks is not an option)_
 - Nested Folders for Bookmarks beneath Collections

### Should Have:
- Movable Global Bookmarks
  - Drag & Drop
  - Remove The Requirement that the referenced collection must be the parent in the app.
  - Optional? Disable Alphabetical auto sorting of bookmarks
- Sharable Bookmark Folders based on user/user role

### Could Have:
- Likely a completely seperate feature request, But the ability to aesthetically hide the base collection in the app, and only have views as starter options for users.
  - **extended:** restrict list views of relational data from even showing up in the navigation for users, while keeping their ability to add to those collections via relational interfaces. (even keep hidden on right click show hidden collections)

## Drawbacks

From a user perspective this is probably less of a new feature, and more of an enhancement... but i have a sneaking suspicion that this might/could be a breaking change/or at least a minor overhaul, if bookmarks are handled the way I think they are, I am not familiar enough with that part of the codebase to know wether or not it is though

## Alternatives
_Creating an extension to replace bookmark functionality is my planned partial short term solution_

## Adoption Strategy
_Possibly a breaking change_

## Unresolved Questions

**Comment by panos-stavrianos** [comment]
We also need exactly that, we currently generate bookmarks programmatically in our app (CRM & Project Management) and after a while it's hard to navigate, especially for managers that have full access. 
I could easily see this implemented in the future as it's really the next logical step, directus already have nested collections and folders.

Trying to find even a _quick & dirty_ workaround I create a folder (`open_projects`) and add a bookmark to it. 
When i navigate into the bookmark i can see that it tries to get `/items/open_projects`
So if there was a way to add something like an `alias` to the actual collection, it could work, i tried to create an sql view but no luck. 
Even if I could add a hook like `open_projects.items.read` and manually return the results it could be something...

@JugaadiTek How do you plan to develop the extension? A full-blown module? Don't you'll have to replicate basically all the IU's functionality (tables, edits etc)?

Any ideas, suggestions?

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
